src/netlink_namespace.c

```diff
@@ -0,0 +1,865 @@
+    if (fields_num != 1) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`%s'. Got %i, expected 1.",
+            key, fields_num);
+      status = -1;
+    } else {
+      add_ignorelist(fields[0], "interface", NULL);
+      if (strcasecmp(key, "VerboseInterface") == 0)
+        add_ignorelist(fields[0], "if_detail", NULL);
+      status = 0;
+    }
+  } else if ((strcasecmp(key, "QDisc") == 0) ||
+             (strcasecmp(key, "Class") == 0) ||
+             (strcasecmp(key, "Filter") == 0)) {
+    if ((fields_num < 1) || (fields_num > 2)) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`%s'. Got %i, expected 1 or 2.",
+            key, fields_num);
+      return (-1);
+    } else {
+      add_ignorelist(fields[0], key, (fields_num == 2) ? fields[1] : NULL);
+      status = 0;
+    }
+  } else if (strcasecmp(key, "IgnoreSelected") == 0) {
+    if (fields_num != 1) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`IgnoreSelected'. Got %i, expected 1.",
+            fields_num);
+      status = -1;
+    } else {
+      if (IS_TRUE(fields[0]))
+        ir_ignorelist_invert = 0;
+      else
+        ir_ignorelist_invert = 1;
+      status = 0;
+    }
+  }
+  else if (strcasecmp(key, "NSName") == 0) {
+    INFO ("netlink_namespace plugin: recevied NSName value %s", fields[0]);
+    if ((fields_num < 1) || (fields_num > 2)) {
+      ERROR ("netlink_namespace plugin: Invalid number of fields for option "
+          "`%s'. Got %i, expected 1 or 2.", key, fields_num);
+      status = -1;
+    } else {
+      strncat(nspath, fields[0], sizeof(nspath) - strlen(nspath));
+      if (access(nspath, F_OK ) != 0 ) {
+        ERROR ("netlink_namespace plugin: ir_config: Unable to access file '%s'.", nspath);
+        status = -1;
+      } else {
+        INFO ("netlink_namespace plugin: ir_config: Setting config to %s", nspath);
+        if ((ns_name = strndup(nspath, sizeof(nspath))) < 0) {
+          ERROR ("netlink_namespace plugin: ir_config: Error while copying/malloc namespace path");
+          status = -1;
+        }
+        INFO ("netlink_namespace pliugin: ir_config: ns_name is now %s", ns_name);
+        status = 0;
+      }
+    }
+  }
+
+  sfree(new_val);
+
+  return (status);
+} /* int ir_config */
+
+static void ir_init_error(int* selfns, int* targetns) {
+  if ((*targetns >= 0) && (close(*targetns) != 0)) {
+    ERROR("Closing \"targetns\" file failed: %m");
+  }
+  if (*selfns >= 0) {
+    if (setns(*selfns, CLONE_NEWNET) != 0) {
+      ERROR("netlink_namespace plugin: ir_init_error: Getting back to original namespace failed: %m");
+    }
+    if  (close(*selfns) != 0) {
+      ERROR("netlink_namespace plugin: ir_init_error: Closing \"selfns\" file failed: %m");
+    }
+  }
+  nltarget = NULL;
+} /* ir_init_error */
+
+static int ir_init (void)
+{
+  int selfns, targetns = -1;
```

Comment:
Since you ascribe special meaning for selfns and targetns being -1, you should probably initialize both to -1. Right now, you only initialize targetns to -1.` int selfns = -1, targetns = -1`